src/mgmt/srv_pool.c

```diff
@@ -433,7 +432,7 @@ ds_mgmt_pool_target_update_state(uuid_t pool_uuid, d_rank_list_t *svc_ranks,
 		 * than allocating an actual list array and populating it
 		 */
 		reint_ranks.rl_nr = 1;
-		reint_ranks.rl_ranks = &rank;
+		reint_ranks.rl_ranks = &target_addrs->pta_addrs[0].pta_rank;
```

Comment:
If you refresh this patch, might be worthwhile to add a comment for this function, stating that target_addrs must only include targets belonging to just one rank.